### PR TITLE
twig: Added overload to Template.render when allow_async=false

### PR DESCRIPTION
**Pull request by dmelcer9**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/twigjs/twig.js/blob/51544ce66e355cf7b142d7f58d52f71088548bad/ASYNC.md#switching-render-mode

### CODE REVIEW

File: types/twig/index.d.ts [hunk 1]
```diff
@@ -24,6 +25,7 @@ export interface Parameters {
 
 export interface Template {
     reset(blocks: any): void;
+    render(context?: any, params?: any, allow_async?: false): string;
     render(context?: any, params?: any, allow_async?: boolean): string | Promise<string>;
```
Comment (enko) [inline types/twig/index.d.ts]: @dmelcer9 wouldn't it be better to explicit define `false` and `true`? `false` returns `string` and `true` returns `Promise<string>` and `undefined` whatever the standard is.

**Review by enko — CHANGES_REQUESTED** [review]
Please have a look at my comment.

File: types/twig/index.d.ts (hunk 1 above)
Comment (dmelcer9) [reply types/twig/index.d.ts]: Does true always return a promise? The name of the allow_async parameter is somewhat ambiguous about this. In any case, we still need the general Boolean overload for when the parameter is unknown at compile time.

Comment (dmelcer9) [reply types/twig/index.d.ts]: Bumping so @typescript-bot doesn't automatically close.

Comment (enko) [reply types/twig/index.d.ts]: @dmelcer9 Sorry did have time to look into this and @typescript-bot seems to be very strict. Maybe we merge as it is. I looked at the code and it wasn't very clear to me, how the default is computed.

**Comment by enko** [comment]
@typescript-bot lgtm

**Comment by dmelcer9** [comment]
I created https://github.com/DefinitelyTyped/DefinitelyTyped/pull/42593 since there doesn't seem to be a way to get typescript-bot to open this PR.

**Pull request closed without merging**